cli/cmd/generate.go

```diff
@@ -529,9 +529,18 @@ func getCombinedYAML(paths []string) ([]byte, error) {
 		if err != nil {
 			return nil, fmt.Errorf("read %s: %w", path, err)
 		}
+		kubeObjs, err := kuberesource.UnmarshalApplyConfigurations(data)
+		if err != nil {
+			return nil, fmt.Errorf("unmarshal %s: %w", path, err)
+		}
+		resource, err := kuberesource.EncodeResources(kubeObjs...)
```

Comment:
Does this also reproduce the `---` doc separators?

Reply:
Yes, they are manually added during the execution of `EncodeResources`. (Actually, I'm pretty sure I saw somewhere in the docs for yaml.v3 that this should also work automatically, so the code there can probably be simplified further)